### PR TITLE
test: implement logic-level unit tests for game core functionality (yachifeng_unit_tests)

### LINKED ISSUE #29: Task: Implement logic-level unit tests for game core functionality (yachifeng_unit_tests)
Performance Rating Logic: Create a test suite to verify the scoring thresholds and player ranking categories (Excellent, Good, Keep Practicing).

Answer Validation Logic: Create a test suite to verify the matching mechanism between user input and correct answers, ensuring case sensitivity and null safety are handled correctly.

### PR DESCRIPTION
**Pull request by yachifeng**
This PR introduces logic-level unit testing to the project, focusing on verifying core game mechanics independently of database operations or UI components. These tests ensure that the mathematical and conditional logic of the game remains robust and accurate.

GameRankLogicTest.java:
- Validates the performance rating system (Excellent, Good, Keep Practicing).
- Tests various score thresholds to ensure players receive the correct rank based on their final score.

AnswerValidationTest.java:
- Validates the answer-matching logic.
- Ensures case sensitivity is handled correctly.
- Implements null-safety checks to prevent the game from crashing when receiving unexpected inputs.

Closes #29

### CODE REVIEW

File: src/test/java/org/example/triviagame/GameRankLogicTest.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+     * Verifies that the rating logic correctly categorizes scores into
+     * Excellent, Good, and Keep Practicing based on predefined thresholds.
+     */
+    @Test
+    public void testGetPerformanceRating() {
+        // Logic thresholds: >= 80: Excellent, 50-79: Good, <50: Keep Practicing
+
+        // Test high score
+        assertEquals("Excellent", getRating(90), "Scores 80 and above should be Excellent.");
+
```
Comment (Diegocastee) [inline src/test/java/org/example/triviagame/GameRankLogicTest.java:27]: I would add a few extra test to look the boundaries:

assertEquals("Excellent", getRating(80), "Score of exactly 80 should be Excellent.");

File: src/test/java/org/example/triviagame/GameRankLogicTest.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        // Test mid-range score
+        assertEquals("Good", getRating(70), "Scores between 50 and 79 should be Good.");
+
```
Comment (Diegocastee) [inline src/test/java/org/example/triviagame/GameRankLogicTest.java:32]: assertEquals("Good", getRating(79), "Score just below 80 should be Good.");
assertEquals("Good", getRating(50), "Score of exactly 50 should be Good.");

File: src/test/java/org/example/triviagame/GameRankLogicTest.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        // Test low score
+        assertEquals("Keep Practicing", getRating(30), "Scores below 50 should be Keep Practicing.");
+    }
```
Comment (Diegocastee) [inline src/test/java/org/example/triviagame/GameRankLogicTest.java:36]: assertEquals("Keep Practicing", getRating(49), "Score just below 50 should be Keep Practicing.");

**Review by Diegocastee — APPROVED** [review]
Good tests

**Comment by yachifeng** [comment]
Thanks for the feedback!  I've updated the test case to include exactly 80, 50, and the thresholds just below them (79 and 49).
